x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4

```diff
@@ -322,3 +323,19 @@ lookupCommand
 inlinestatsCommand
     : DEV_INLINESTATS stats=aggFields (BY grouping=fields)?
     ;
+
+joinCommand
+    : type=(DEV_JOIN_LOOKUP | DEV_JOIN_LEFT | DEV_JOIN_RIGHT)? DEV_JOIN joinTarget joinCondition
+    ;
+
+joinTarget
+    : index=identifier (AS alias=identifier)?
```

Comment:
I've added this to the metadata https://github.com/elastic/elasticsearch/issues/116208
It will be delicate since the command already uses identifiers which will clashes/is ambiguous with the indexSource declaration.